ci/upload_cudf_java_docs.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+validate_version() {
+    local version=$1
+    if ! [[ $version =~ ^[0-9]{2}\.[0-9]{2}(\.[0-9]+)?$ ]]; then
+        echo "Error: Version must be in format YY.MM or YY.MM.P" >&2
+        return 1
+    fi
+}
+
+cleanup() {
+    rm -rf cudf-*-javadoc.jar cudf-docs 2>/dev/null || true
```

Comment:
```suggestion
    rm -rf cudf-*-javadoc.jar cudf-docs
```

Is there any reason to include these last few commands/operators?

The `-f` flag should prevent the need to use `|| true`.